deps/blastrampoline.mk

```diff
@@ -9,23 +9,33 @@ $(eval $(call git-external,blastrampoline,BLASTRAMPOLINE,,,$(BUILDDIR)))
 BLASTRAMPOLINE_BUILD_OPTS := $(MAKE_COMMON) CC="$(CC) $(SANITIZE_OPTS)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)"
 BLASTRAMPOLINE_BUILD_OPTS += ARCH="$(ARCH)" OS="$(OS)"
 
-$(BUILDDIR)/$(BLASTRAMPOLINE_SRC_DIR)/build-configured: $(BUILDDIR)/$(BLASTRAMPOLINE_SRC_DIR)/source-extracted
+BLASTRAMPOLINE_BUILDDIR := $(BUILDDIR)/$(BLASTRAMPOLINE_SRC_DIR)
+
+$(BLASTRAMPOLINE_BUILDDIR)/blastrampoline-cygwin_build_fix.patch-applied: $(BLASTRAMPOLINE_BUILDDIR)/source-extracted
+	cd $(BLASTRAMPOLINE_BUILDDIR) && \
+		patch -p1 -f < $(SRCDIR)/patches/blastrampoline-cygwin_build_fix.patch
+	echo 1 > $@
+
+$(BLASTRAMPOLINE_BUILDDIR)/source-patched: $(BLASTRAMPOLINE_BUILDDIR)/blastrampoline-cygwin_build_fix.patch-applied
+	echo 1 > $@
+
+$(BUILDDIR)/$(BLASTRAMPOLINE_SRC_DIR)/build-configured: $(BLASTRAMPOLINE_BUILDDIR)/source-patched
 	mkdir -p $(dir $@)
 	echo 1 > $@
 
 BLASTRAMPOLINE_BUILD_ROOT := $(BUILDDIR)/$(BLASTRAMPOLINE_SRC_DIR)/src
 $(BUILDDIR)/$(BLASTRAMPOLINE_SRC_DIR)/build-compiled: $(BUILDDIR)/$(BLASTRAMPOLINE_SRC_DIR)/build-configured
 	cd $(dir $@)/src && $(MAKE) $(BLASTRAMPOLINE_BUILD_OPTS)
-ifeq ($(OS), WINNT)
-	# Windows doesn't like soft link, use hard link
-	cd $(BLASTRAMPOLINE_BUILD_ROOT)/build/ && \
-		cp -f --dereference --link libblastrampoline.dll libblastrampoline.dll
-endif
 	echo 1 > $@
 
 define BLASTRAMPOLINE_INSTALL
 	$(MAKE) -C $(BLASTRAMPOLINE_BUILD_ROOT) install $(BLASTRAMPOLINE_BUILD_OPTS) DESTDIR="$2"
 endef
+ifeq ($(OS), WINNT)
+# Windows doesn't like soft link, use hard link to copy file without version suffix
+BLASTRAMPOLINE_INSTALL += && cd $2$$(build_prefix)/bin && \
+$$(WIN_MAKE_HARD_LINK) libblastrampoline-*.dll libblastrampoline.dll
```

Comment:
Do we really need `libblastrampoline.dll`?
Maybe we don't need it anymore:

https://github.com/JuliaLang/julia/blob/5f8fcd9423685102640b2b37b16c34877241b92b/stdlib/libblastrampoline_jll/src/libblastrampoline_jll.jl#L22-L24

Reply:
We can instead fix the fact that LIBSUITESPARSE wants it.

Tim S.

Reply:
I have decided that fixing LIBSUITESPARSE is beyond me at this time; do you wish me to remove the WIN_MAKE_HARD_LINK code from this PR.

Reply:
> I have decided that fixing LIBSUITESPARSE is beyond me at this time

maybe
```diff
diff --git a/deps/libsuitesparse.mk b/deps/libsuitesparse.mk
index 7d79e03ee8..00e1776af3 100644
--- a/deps/libsuitesparse.mk
+++ b/deps/libsuitesparse.mk
@@ -24,8 +24,8 @@ SUITESPARSE_LIB += $(RPATH_ESCAPED_ORIGIN)
 endif
 LIBSUITESPARSE_MFLAGS := CC="$(CC) $(SANITIZE_OPTS)" CXX="$(CXX) $(SANITIZE_OPTS)" F77="$(FC)" \
          AR="$(AR)" RANLIB="$(RANLIB)" \
-         BLAS="-L$(build_shlibdir) -lblastrampoline" \
-         LAPACK="-L$(build_shlibdir) -lblastrampoline" \
+         BLAS="-L$(build_shlibdir) -lblastrampoline-5" \
+         LAPACK="-L$(build_shlibdir) -lblastrampoline-5" \
          LDFLAGS="$(SUITESPARSE_LIB) $(SANITIZE_LDFLAGS)" CFOPENMP="" CUDA=no CUDA_PATH="" \
          UMFPACK_CONFIG="$(UMFPACK_CONFIG)" \
          CHOLMOD_CONFIG="$(CHOLMOD_CONFIG)" \
```
https://github.com/JuliaPackaging/Yggdrasil/blob/7b3cd7ff394c782ec9f0e7e6cda571316d077835/S/SuiteSparse/SuiteSparse/build_tarballs.jl#L33-L35

I will try to fix this in the new pr.